RedissonAutoConfiguration fails to convert nodes in Native Image (GraalVM) application

### Redis version

7.2.4

### Redisson version

3.52.0

### Redisson configuration

Standard Spring config:
```
spring:
  data:
    redis:
      cluster:
        nodes: [127.0.0.1:6379]
```

### What is the Expected behavior?

_No response_

### What is the Actual behavior?

`convertNodes(String prefix, List<Object> nodesObject)` in the auto configuration class takes nodes as `List<Object>` even though in both places it's invoked from, it's the `List<RedisConnectionDetails.Node>`. 

The rationale behind this might be multi-version support, but it was added in #5273 for Spring Boot 3.1, which had the same `record Node(String host, int port)` since its introduction.

But because of that, it tries to use reflection over the class and get `host` and `port` fields, and fails at runtime when running as a GraalVM native image executable:
```
Caused by: java.lang.NullPointerException
	at org.springframework.util.ReflectionUtils.makeAccessible(ReflectionUtils.java:800)
	at org.redisson.spring.starter.RedissonAutoConfiguration.convertNodes(RedissonAutoConfiguration.java:384)
	at org.redisson.spring.starter.RedissonAutoConfiguration.redisson(RedissonAutoConfiguration.java:279)
	at org.redisson.spring.starter.RedissonAutoConfigurationV2__BeanDefinitions.lambda$getRedissonInstanceSupplier$0(RedissonAutoConfigurationV2__BeanDefinitions.java:137)
```

A workaround would be to create a custom RedissonClient tailored to my needs, without using reflection, etc., but this somewhat defeats the purpose of auto-configuration. Unless I'm missing something, accepting `List<Node>` and returning `node.host + ":" + node.port` should be enough. 

### Additional information

_No response_

--- a/redisson-spring/redisson-spring-boot-starter/src/main/java/org/redisson/spring/starter/RedissonAutoConfiguration.java
+++ b/redisson-spring/redisson-spring-boot-starter/src/main/java/org/redisson/spring/starter/RedissonAutoConfiguration.java
@@ -367,6 +367,7 @@ public class RedissonAutoConfiguration {
         return prefix;
     }
 
+    @SuppressWarnings("IllegalCatch")
     private String[] convertNodes(String prefix, List<?> nodesObject) {
         List<String> nodes = new ArrayList<>(nodesObject.size());
         try {
@@ -382,7 +383,7 @@ public class RedissonAutoConfiguration {
 
                 nodes.add(prefix + host + ":" + port);
             }
-        } catch (Exception e) {
+        } catch (Throwable e) {
             throw new IllegalStateException("Failed to convert nodes", e);
         }
         return nodes.toArray(new String[0]);

--- a/redisson-spring/redisson-spring-boot-starter/src/main/java/org/redisson/spring/starter/RedissonAutoConfigurationV4.java
+++ b/redisson-spring/redisson-spring-boot-starter/src/main/java/org/redisson/spring/starter/RedissonAutoConfigurationV4.java
@@ -334,6 +334,7 @@ public class RedissonAutoConfigurationV4 {
         return prefix;
     }
 
+    @SuppressWarnings("IllegalCatch")
     private String[] convertNodes(String prefix, List<?> nodesObject) {
         List<String> nodes = new ArrayList<>(nodesObject.size());
         try {
@@ -349,7 +350,7 @@ public class RedissonAutoConfigurationV4 {
 
                 nodes.add(prefix + host + ":" + port);
             }
-        } catch (Exception e) {
+        } catch (Throwable e) {
             throw new IllegalStateException("Failed to convert nodes", e);
         }
         return nodes.toArray(new String[0]);

